src/SIS_sponge.F90

```diff
@@ -0,0 +1,991 @@
+!> Implements relaxation regions in SIS2 model
+!> for sea ice concentration (partial area) and thickness by categories
+!> The algorithm can be used to impose open boundary conditions 
+!> for sea ice thickness and partial area in regional SIS2 applications, 
+!> or for correcting the fields within the domain.
+!>
+!> Dmitry Dukhovskoy NOAA OAR PSL 2025
+!> 
+module SIS_sponge
+
+!! Module to read in Time of the ice fields, ice concentration, ice thickness, relaxation time scale
+use MOM_coms,          only : sum_across_PEs, max_across_PEs
+use MOM_coms,          only : PE_here   !! debugging
+use MOM_time_manager,  only : time_type, set_date, get_time, get_date
+use MOM_unit_scaling,  only : unit_scale_type
+use ice_grid,          only : ice_grid_type
+
+use MOM_error_handler, only : SIS_error=>MOM_error, FATAL, WARNING, SIS_mesg=>MOM_mesg
+use MOM_error_handler, only : MOM_get_verbosity
+use MOM_file_parser,   only : get_param, log_param, log_version, param_file_type
+use MOM_io,            only : file_exists, MOM_read_data, slasher
+use MOM_io,            only : axis_info
+use MOM_interpolate,   only : init_external_field, get_external_field_info, time_interp_external_init
+use MOM_interpolate,   only : time_interp_external
+use MOM_interpolate,   only : external_field
+
+use SIS_diag_mediator, only : post_SIS_data, post_data=>post_SIS_data
+use SIS_diag_mediator, only : register_diag_field=>register_SIS_diag_field
+use SIS_hor_grid,      only : SIS_hor_grid_type
+use SIS_sum_output,    only : SIS_sum_out_CS, write_ice_statistics! , SIS_sum_output_init
+use SIS_types,         only : ice_state_type, IST_chksum, IST_bounds_check, total_sfc_flux_type
+use SIS_types,         only : ocean_sfc_state_type, ice_ocean_flux_type, fast_ice_avg_type
+use SIS2_ice_thm,      only : SIS2_ice_thm_CS, SIS2_ice_thm_init, SIS2_ice_thm_end
+use SIS2_ice_thm,      only : get_SIS2_thermo_coefs, enthalpy_liquid_freeze
+use SIS2_ice_thm,      only : enth_from_TS, Temp_from_En_S, enthalpy_liquid, calculate_T_freeze
+use SIS_optics,        only : VIS_DIR, VIS_DIF, NIR_DIR, NIR_DIF    ! debugging only, delete later
+use SIS_utils,         only : is_NaN
+
+implicit none; private
+
+#include <SIS2_memory.h>
+
+public initialize_icerelax_file, apply_isponge, set_up_isponge_field, SIS_sponge_end
+public global_to_local_ij
```

Comment:
I do not believe that `global_to_local_ij` is used outside of this module. If so it need not be declared as public.